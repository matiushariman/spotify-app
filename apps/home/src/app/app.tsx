import NxWelcome from './nx-welcome';

export function App() {
  return (
    <>
      <NxWelcome title="home" />
      <div />
    </>
  );
}

export default App;
