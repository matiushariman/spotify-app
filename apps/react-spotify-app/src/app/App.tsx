import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import { routes as authRoutes } from '@react-spotify/auth-feature';
import { useSessionStore } from '@react-spotify/shared-stores';
import { useMemo } from 'react';

export function App() {
  const isAuthenticated = useSessionStore((state) => state.isAuthenticated);
  const router = useMemo(
    () => createBrowserRouter([...authRoutes]),
    [isAuthenticated]
  );

  return <RouterProvider router={router} />;
}

export default App;
