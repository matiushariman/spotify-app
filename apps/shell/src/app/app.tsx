import { useEffect, Suspense, lazy } from 'react';
import { Route, Routes, useLocation } from 'react-router-dom';
import { AuthUtils } from '@spotify-app/auth-utils';
import { Authorization } from '@spotify-app/api-client';

import { Callback } from './pages/Callback';

const Home = lazy(() => import('home/Module'));

const authUtils = new AuthUtils();
const authorization = new Authorization({
  clientId: process.env['NX_SPOTIFY_CLIENT_ID'],
});

export function App() {
  const { pathname } = useLocation();
  const navigateToSpotify =
    !authUtils.checkToken() && !pathname.includes('callback');

  useEffect(() => {
    if (navigateToSpotify) {
      window.location.replace(authorization.getAuthorizationUrl());
    }
  }, [navigateToSpotify]);

  if (navigateToSpotify) {
    return null;
  }

  return (
    <Suspense fallback={null}>
      <Routes>
        <Route path="/callback" element={<Callback />} />
        <Route path="/" element={<Home />} />
      </Routes>
    </Suspense>
  );
}

export default App;
