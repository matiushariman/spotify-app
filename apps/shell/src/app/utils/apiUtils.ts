import type { ErrorResponse } from '@spotify-app/api-client';
import { AuthUtils } from '@spotify-app/auth-utils';

export const handleApiError = (err: ErrorResponse) => {
  const {
    error: { status },
  } = err;

  if (status === 401) {
    window.alert(
      'Your session has expired! Click close to reauthenticate your session.'
    );
    const authUtils = new AuthUtils();
    authUtils.removeToken();

    window.location.replace('/');
  }
};
