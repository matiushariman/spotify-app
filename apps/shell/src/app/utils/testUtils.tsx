import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import {
  RouterProvider,
  createMemoryRouter,
  RouteObject,
} from 'react-router-dom';
import { QueryClientProvider, QueryClient } from 'react-query';

import '../../styles.css';

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  const routes: RouteObject[] = [
    {
      path: '/',
      element: children,
    },
  ];
  const router = createMemoryRouter(routes);

  return (
    <QueryClientProvider client={new QueryClient()}>
      <RouterProvider router={router} />
    </QueryClientProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };
