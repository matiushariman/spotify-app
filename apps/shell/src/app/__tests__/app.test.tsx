import { App } from '../app';
import { render } from '../utils/testUtils';

const mockWindowReplace = jest.fn();

Object.defineProperty(window, 'location', {
  value: {
    replace: mockWindowReplace,
  },
});

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useLocation: jest
    .fn()
    .mockReturnValueOnce({
      pathname: {
        includes: () => false,
      },
    })
    .mockReturnValueOnce({
      pathname: {
        includes: () => true,
      },
    }),
}));

jest.mock('@spotify-app/auth-utils', () => ({
  AuthUtils: jest
    .fn()
    .mockReturnValueOnce({
      checkToken: () => false,
    })
    .mockReturnValueOnce({
      checkToken: () => true,
    }),
}));

describe('App', () => {
  it('should render App that matches snapshot when navigateToSpotify is false', () => {
    const { baseElement } = render(<App />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should call window.location.replace when navigateToSpotify is true', () => {
    render(<App />);

    expect(mockWindowReplace).toHaveBeenCalled();
  });
});
