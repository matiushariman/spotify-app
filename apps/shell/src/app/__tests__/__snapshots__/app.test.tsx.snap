// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App that matches snapshot when navigateToSpotify is false 1`] = `
<body>
  <div />
</body>
`;
